cli/src/cmd_api.rs

```diff
@@ -148,19 +148,25 @@ impl crate::AuthenticatedCmd for CmdApi {
 
             // Set this as our body.
             bytes = buf.clone();
+        }
 
-            // Set our params to the query string.
-            if !params.is_empty() {
-                let mut query_string = String::new();
-                for (key, value) in params {
-                    if !query_string.is_empty() {
-                        query_string.push('&');
-                    }
+        // Set our params to the query string.
+        if !params.is_empty() {
+            let mut query_string = String::new();
+            for (key, value) in params {
+                if !query_string.is_empty() {
+                    query_string.push('&');
+                }
+                if value.is_string() {
+                    // Avoid injecting quotes around string values by using `Value::as_str()`.
+                    // Unwrap is safe as we have just confirmed this is a `Value::String`.
+                    query_string.push_str(&format!("{}={}", key, value.as_str().unwrap()));
+                } else {
                     query_string.push_str(&format!("{}={}", key, value));
                 }
```

Comment:
What is it when it's not a string?

Reply:
@david-crespo with the `-F/--field` option, we may parse the value to the `Null`, `Bool`, and `Number` variants of `Value`. These all have straightforward string representations, though, and can be appended directly to the parameters.

```sh
$ oxide api /v1/disks -F project=1.0
uri = "https://oxide.sys.example.com/v1/disks?project=1.0"
```

Reply:
Should we `match value` and complain if it's not a scalar value?

Reply:
I wondered the same thing, so I say yes

Reply:
Good call!